torch/csrc/autograd/record_function_ops.h

```diff
@@ -24,4 +26,29 @@ TORCH_API c10::intrusive_ptr<c10::ivalue::Future> _call_end_callbacks_on_fut_new
     const c10::intrusive_ptr<PythonRecordFunction>& record,
     const c10::intrusive_ptr<c10::ivalue::Future>& fut);
 
+/**
+ * RAII wrapper that behaves similarly to torch.profiler.record_function.
+ */
+class RecordFunction {
```

Comment:
Isn't this part of libtorch_python? Would nativert depend on libtorch-only or libtorch_python?

Reply:
Ok, let's add a test to make sure all of this behave as expected